src/service/topics.js

```diff
@@ -14,9 +15,14 @@ const getUserTopic = async (query) => {
   }
 };
 
-const putUserTopic = async (data) => {
+const putUserTopic = async (data, isIncognito) => {
+  let res;
   try {
-    const res = await api.put('/topics/follow-v2', data);
+    if (isIncognito) {
+      res = await anonymousApi.put('/topics/follow-v2', data);
+    } else {
+      res = await api.put('/topics/follow-v2', data);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are good as they allow for handling both regular and incognito users. However, there is no error handling for the case when `isIncognito` is not provided or is not a boolean. It would be better to add a check for this to prevent potential issues.
```diff
  const putUserTopic = async (data, isIncognito) => {
    let res;
    try {
+     if (typeof isIncognito !== 'boolean') {
+       throw new Error('Invalid value for isIncognito');
+     }
      if (isIncognito) {
        res = await anonymousApi.put('/topics/follow-v2', data);
      } else {
        res = await api.put('/topics/follow-v2', data);
      }
      OneSignalUtil.rebuildAndSubscribeTags();
      return res.data;
    } catch (error) {
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->